frontend/src/pages/projects/screens/detail/storage/ClusterStorageTableRow.tsx

```diff
@@ -0,0 +1,289 @@
+import {
+  ActionList,
+  ActionListItem,
+  Button,
+  HelperText,
+  HelperTextItem,
+  InputGroup,
+  InputGroupItem,
+  InputGroupText,
+  List,
+  ListItem,
+  Popover,
+  Stack,
+  StackItem,
+  TextInput,
+  Bullseye,
+  Spinner,
+  Label,
+} from '@patternfly/react-core';
+import { MinusCircleIcon } from '@patternfly/react-icons';
+import { Td, Tr } from '@patternfly/react-table';
+import React from 'react';
+import TypeaheadSelect from '~/components/TypeaheadSelect';
+import { NotebookKind, PersistentVolumeClaimKind } from '~/k8sTypes';
+import { ClusterStorageNotebookSelection, MountPath } from '~/pages/projects/types';
+import { MOUNT_PATH_PREFIX } from '~/pages/projects/screens/spawner/storage/const';
+import SimpleSelect from '~/components/SimpleSelect';
+import { MountPathFormat } from '~/pages/projects/screens/spawner/storage/types';
+import { getNotebookPVCMountPathMap } from '~/pages/projects/notebook/utils';
+import {
+  getInUseMountPaths,
+  isMountPathFormat,
+  mountPathFormat,
+  mountPathSuffix,
+  validateClusterMountPath,
+} from './utils';
+
+type ClusterStorageTableRowProps = {
+  obj: ClusterStorageNotebookSelection;
+  connectedNotebooks: NotebookKind[];
+  existingPvc?: PersistentVolumeClaimKind;
+  clusterStorageName: string;
+  addNotebookData: (
+    notebookName?: string,
+    mountPath?: MountPath,
+    updatedValue?: boolean,
+    existingPvc?: boolean,
+  ) => void;
+  availableNotebooks: { notebooks: NotebookKind[]; loaded: boolean };
+  onDelete: () => void;
+};
+
+const ClusterStorageTableRow: React.FC<ClusterStorageTableRowProps> = ({
+  obj,
+  connectedNotebooks,
+  existingPvc,
+  clusterStorageName,
+  addNotebookData,
+  availableNotebooks,
+  onDelete,
+}) => {
+  const inUseMountPaths = React.useMemo(
+    () => getInUseMountPaths(obj.name, availableNotebooks.notebooks, existingPvc?.metadata.name),
+    [availableNotebooks.notebooks, existingPvc?.metadata.name, obj.name],
+  );
+
+  React.useEffect(() => {
+    if (!obj.existingPvc && !obj.isUpdatedValue) {
+      const defaultPathValue = clusterStorageName
+        ? `${MOUNT_PATH_PREFIX}${clusterStorageName.toLowerCase().replace(/\s+/g, '-')}-${
+            obj.newRowId ?? 1
+          }`
+        : '';
+      addNotebookData(
+        undefined,
+        clusterStorageName
+          ? {
+              value: defaultPathValue,
+              error: validateClusterMountPath(defaultPathValue, inUseMountPaths),
+            }
+          : undefined,
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [clusterStorageName, inUseMountPaths, obj.existingPvc, obj.isUpdatedValue, obj.newRowId]);
```

Comment:
why do this in the row. can you just do it in the component above where you are adding the row

Reply:
@Gkrumbach07 since I am getting the `clusterStorageName` from [BaseStorageModal](https://github.com/opendatahub-io/odh-dashboard/pull/3529/files/18ab33f4ccab30a4b5ef5618f2df2a32b1cbc160#diff-4f31835a5a1bd8e7d3f245e9a812610197f3e1e9cca51f44e9560a227130057d) and the name changes upon typing , I prefered to keep it in the row  name  as this cannot be treated as default `default` to add while creating row